misc/python/materialize/cli/ci_coverage_pr_report.py

```diff
@@ -0,0 +1,90 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+import re
+import subprocess
+import sys
+from collections import OrderedDict
+from itertools import groupby
+from typing import Dict, List
+
+Coverage = Dict[str, OrderedDict[int, int]]
+SOURCE_RE = re.compile(
+    "^/var/lib/buildkite-agent/builds/buildkite-.*/materialize/tests/(.*$)"
+)
+
+
+def find_interesting_lines() -> Coverage:
+    coverage: Coverage = {}
+    # TODO: Get base state from buildkite instead of just HEAD~
+    result = subprocess.run(
+        ["git", "diff", "-U0", "HEAD~"], check=True, capture_output=True
+    )
+    file = None
+    for line_raw in result.stdout.splitlines():
+        line = line_raw.decode("utf-8")
+        if line.startswith("+++"):
+            if not line.endswith(".rs"):
+                continue
+            file = line.removeprefix("+++ b/")
+            coverage[file] = OrderedDict()
+        elif line.startswith("@@ ") and file:
+            parts = line.split(" ")[2]
+            if "," in parts:
+                start, length = map(int, parts.split(","))
+            else:
+                start = int(parts)
+                length = 1
+            for line_nr in range(start, start + length):
+                coverage[file][line_nr] = 0
+    return coverage
+
+
+def mark_covered_lines(lcov_file: str, coverage: Coverage) -> None:
+    for line in open(lcov_file):
```

Comment:
A comment with an example fragment of the file being parsed here or a pointer to the specification of the format would be nice here.